public/app/themes/clarity/src/components/c-intranet-switcher/view.php

```diff
@@ -7,6 +7,13 @@
 $current_intranet = get_intranet_code();
 
 $referrer = (isset($_SERVER['HTTP_REFERER']) ? parse_url($_SERVER['HTTP_REFERER']) : parse_url(get_home_url()));
+$url = get_home_url();
+
+// Redirect back to the previous page after selecting an agency if this is their first visit
+if (isset($_GET['send_back'])) {
+    $url = $_GET['send_back'];
```

Comment:
We could end up with some issues here if we don't do validation.

For example a bad link could be shared to redirect people to a phishing site.

http://intranet.docker/agency-switcher/?send_back=http%3A%2F%2Fgoogle.com%2Fnewspage%2F

One way to prevent this would be for `send_back` to be checked that it starts with the home url (including a trailing slash - else the malicious url could be https://intranet.justice.gov.uk.haxor.com) .

Reply:
@EarthlingDavey thanks for that, good spot. I've added an additional `wp_validate_redirect` check.

Reply:
@EmilyHazlehurst that's awesome! It's good to know about this wp function, as I've could have used it a few times.